lib/services/player_screen_theme_provider.dart

```diff
@@ -1,39 +1,69 @@
 import '../generate_material_color.dart';
 import 'current_album_image_provider.dart';
 
-final playerScreenThemeProvider =
-    FutureProvider.family<ColorScheme?,Brightness>((ref,brightness) async {
-        ImageProvider? image = ref.watch(currentAlbumImageProvider);
-        if (image != null) {
-
-            final palette =
-                await PaletteGenerator.fromImageProvider(
-                image,
-                timeout: const Duration(milliseconds: 2000),
-            );
-
-            // Color accent = palette.dominantColor!.color;
-            Color accent = palette.vibrantColor?.color ?? palette.dominantColor?.color ?? const Color.fromARGB(255, 0, 164, 220);
-
-            final lighter = brightness == Brightness.dark;
-
-            final background = Color.alphaBlend(
-                lighter
-                    ? Colors.black.withOpacity(0.675)
-                    : Colors.white.withOpacity(0.675),
-                accent);
-
-            accent = accent.atContrast(4.5, background, lighter);
-
-            return ColorScheme.fromSwatch(
-                    primarySwatch: generateMaterialColor(accent),
-                    accentColor: accent,
-                    brightness: brightness,
-                );
-        }
-    });
+final AutoDisposeProviderFamily<ColorScheme, Brightness> playerScreenThemeProvider =
+Provider.family.autoDispose<ColorScheme, Brightness>((ref, brightness) {
+  ColorScheme? scheme = ref.watch(playerScreenThemeNullableProvider(brightness)).value;
+    if(scheme != null){
+      return scheme;
+  }else{
+      final lighter = brightness == Brightness.dark;
+      Color accent = lighter
+          ? const Color.fromARGB(255, 50, 50, 50)
+          : const Color.fromARGB(255, 200, 200, 200);
+
+      final background = Color.alphaBlend(
+          lighter
+              ? Colors.black.withOpacity(0.675)
+              : Colors.white.withOpacity(0.675),
+          accent);
+
+      accent = accent.atContrast(4.5, background, lighter);
+
+      return ColorScheme.fromSwatch(
+        primarySwatch: generateMaterialColor(accent),
+        accentColor: accent,
+        brightness: brightness,
+      );
+    }
+});
+
+final AutoDisposeFutureProviderFamily<ColorScheme?, Brightness> playerScreenThemeNullableProvider =
+    FutureProvider.family.autoDispose<ColorScheme?, Brightness>((ref, brightness) async {
+
+  ImageProvider? image = await ref.watch(currentAlbumImageProvider.future);
+  if (image==null) return null;
+  Logger("colorProvider").fine("Re-theming based on image $image");
+
+    final PaletteGenerator? palette = await PaletteGenerator.fromImageProvider(
+        image,
+        timeout: const Duration(seconds: 5),
+      ).then((value) => value, onError: (_)=>null);
+    if (palette==null) return null;
+    // Color accent = palette.dominantColor!.color;
+    Color accent = palette.vibrantColor?.color ??
+        palette.dominantColor?.color ??
+        const Color.fromARGB(255, 0, 164, 220);
```

Comment:
Finamp's default blue accent color (`0, 164, 220`) should be used as a fallback. You can basically get rid of lines 16-28 I think, because that code will only run if the previous attempt at color generation failed, and in that case the fallback should be blue, not gray :)  
It also simplifies the code a bit.

Oh and at the end, please format the files you've changed (using your editor or `dart format path/to/file`

Reply:
I did some comparisons of the different default colors.  Do you think the raw blue color is best?

Current color:
![Screenshot (48)](https://github.com/jmshrv/finamp/assets/45665554/337e8213-fa99-46fd-bbcf-a90e6463bdea)
![Screenshot (52)](https://github.com/jmshrv/finamp/assets/45665554/892f35d0-6a99-4cd0-9570-f1ce8120f1fe)

Default blue color with accent calculation steps:
![Screenshot (47)](https://github.com/jmshrv/finamp/assets/45665554/bab87044-9c23-471d-a4e6-0878823085a9)
![Screenshot (51)](https://github.com/jmshrv/finamp/assets/45665554/ba6521ee-7ed3-45ec-9f45-4ff7b947477b)

 return ColorScheme.fromSeed(seedColor: Color.fromARGB(255, 0, 164, 220),brightness: brightness):
![Screenshot (55)](https://github.com/jmshrv/finamp/assets/45665554/40952161-edc4-4048-b2f5-a2749aed781d)
![Screenshot (57)](https://github.com/jmshrv/finamp/assets/45665554/7488d06d-c530-4f25-8118-51696704bd39)

Reply:
Yeah I think it should be fine. But I was thinking more about a generall fallback if a cover doesn't load/exist. The "restoring queue" widget could stay gray if you prefer that :)

Reply:
Also, the contrast checking is not really useful if a cover fails to load, since then the background (blurred cover) and accent color don't match anyway. See example here;

Cover for themeprovider failed to load, blurred cover loaded eventually (notice the low contrast):

![Screenshot_20231213-113154.png](https://github.com/jmshrv/finamp/assets/18015147/ca153048-74d7-410a-b2ac-6df8ec5fdedc)

After restart, when all covers loaded successfully:

![Screenshot_20231213-113221.png](https://github.com/jmshrv/finamp/assets/18015147/1967f96b-1768-4c33-916f-bbe22eeaf904)

Reply:
The theme provider now returns grey for the initial queue loading, but blue if palette generation fails.